examples/seldon_deployment/run.py

```diff
@@ -119,31 +132,31 @@ def main(
     penalty_strength: float,
     toleration: float,
     min_accuracy: float,
-    kubernetes_context: str,
-    namespace: str,
-    base_url: str,
+    secret: str,
     stop_service: bool,
 ):
     """Run the Seldon example continuous deployment or inference pipeline
 
     Example usage:
 
-    python run.py --deploy --predict \
-        --model-flavor tensorflow \
-        --kubernetes-context=zenml-eks-sandbox \
-        --namespace=kubeflow \
-        --base-url=http://abb84c444c7804aa98fc8c097896479d-377673393.us-east-1.elb.amazonaws.com \
-        --min-accuracy 0.80
+        python run.py --deploy --predict --model-flavor tensorflow \
+             --min-accuracy 0.80 --secret seldon-init-container-secret
+
     """
+    model_name = "mnist"
+    deployment_pipeline_name = "continuous_deployment_pipeline"
+    deployer_step_name = "seldon_model_deployer_step"
+
+    model_deployer = SeldonModelDeployer.get_active_model_deployer()
 
     if stop_service:
-        service = load_last_service_from_step(
-            pipeline_name="continuous_deployment_pipeline",
-            step_name="model_deployer",
-            running=True,
+        services = model_deployer.find_model_server(
+            pipeline_name=deployment_pipeline_name,
+            pipeline_step_name=deployer_step_name,
+            model_name=model_name,
         )
-        if service:
-            service.stop(timeout=100)
+        if services:
+            services[0].stop(timeout=100)
```

Comment:
Does this always return a list of len=1 ? If now what are the other entries?

Reply:
The Seldon Core model deployer step guarantees to have at most one service running at any given time for the same pipeline_name, step_name and model_name. So unless you do something sneaky, like create a SeldonDeployment resource in the Kubernetes cluster manually with the same characteristics, you should only expect at most one entry here.